packages/astro/src/cli/add/index.ts

```diff
@@ -72,12 +71,28 @@ const OFFICIAL_ADAPTER_TO_IMPORT_MAP: Record<string, string> = {
 	deno: '@astrojs/deno',
 };
 
+async function detectPackageManager(cwd = process.cwd()) {
+	if (existsSync(path.join(cwd, 'package-lock.json'))) {
+		return { name: 'npm' };
+	}
+	if (existsSync(path.join(cwd, 'yarn.lock'))) {
+		return { name: 'yarn' };
+	}
+	if (existsSync(path.join(cwd, 'pnpm-lock.yaml'))) {
+		return { name: 'pnpm' };
+	}
+	if (existsSync(path.join(cwd, 'bun.lock'))) {
+		return { name: 'bun' };
+	}
+	return null;
+}
```

Comment:
Slightly hesitant about swapping out `preferred-pm` for this custom logic. It's essentially the same but I think `preferred-pm` has some fallbacks that this doesn't cover?

Reply:
I guess if Bun isn't supported by `preferred-pm`, we do kinda need to use our own!

Reply:
I could implement additional checks, and error handling, and utilize which-pm as a fallback.

Reply:
Bun I've submitted a PR to `preferred-pm` but I'm not sure when it'll land